operator/controllers/agent/reconciler.go

```diff
@@ -408,37 +408,29 @@ func (r *AgentReconciler) updateAgent(ctx context.Context, instance *agentv1alph
 
 // checkDefaults checks and sets defaults when the Defaulter webhook is not triggered.
 func (r *AgentReconciler) checkDefaults(ctx context.Context, instance *agentv1alpha1.Agent) error {
-	resource, err := r.DynamicClient.Resource(api.GroupVersion.WithResource("agents")).Namespace(instance.GetNamespace()).Get(ctx, instance.GetName(), v1.GetOptions{})
-	if err != nil {
+	updateStatus := func(instance *agentv1alpha1.Agent, reason, message string) error {
 		instance.Status.Resources = controllers.FailedStatus
-		r.Recorder.Eventf(instance, corev1.EventTypeWarning, "FailedToFetch", "Failed to fetch Resource. Error: '%s'", err.Error())
+		r.Recorder.Eventf(instance, corev1.EventTypeWarning, reason, message)
```

Comment:
Are these changes related?

Reply:
I made it a common function as the same was used multiple times